### DISCUSSION TITLE
Editing Profile

### DISCUSSION THREAD

**Discussion by stuartpy** [AzureAD/microsoft-identity-web#1379] [Q&A]
I am trying to link to the page that lets users update their profile, so in ASP.net Core I set up a link like this
`<a asp-area="MicrosoftIdentity" asp-controller="Account" asp-action="EditProfile">Change these details</a>`

The error I get back is:
An error occurred while processing your request.
Request ID: 00-92c2610424958440ac0e51d6b898c5d8-ca684b1a9f964f43-00

Details
Message contains error: 'invalid_grant', error_description: 'AADB2C90088: The provided grant has not been issued for this endpoint. Actual Value : B2C_1_SignUpIn and Expected Value : B2C_1_ProfileEditing Correlation ID: 574cba99-159b-47c5-b17e-8db3661a14fc Timestamp: 2021-08-11 10:44:41Z ', error_uri: 'error_uri is null'.

I did find this from a year ago https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/issues/399 reporting the same thing, but the issue says fixed.  I am using the latest versions of Identity.Web and Identity.Web.Ui (1.15.2)

I have also cloned a sample project using dotnet new webapp2 --auth  IndividualB2C which contains this code:

```
@if (!string.IsNullOrEmpty(options.EditProfilePolicyId))
        {
            <li class="nav-item">
                <a class="nav-link text-dark" asp-area="MicrosoftIdentity" asp-controller="Account" asp-action="EditProfile">
                    <span class="text-dark">Hello @User.Identity.Name!</span>
                </a>
            </li>
        }
```
However when I run it, options.EditProfilePolicyId is null (as are many of the other options) and if I take this out and just try to go to Edit Profile I am redirected back to the home page.

I'm not sure what to do from here.  Please can someone advise?

Thanks

**Comment by jennyf19** [comment]
@stuartpy is this still showing up with the latest version of id web? 1.18?
